lib/contentTypeParser.js

```diff
@@ -184,6 +184,7 @@ ContentTypeParser.prototype.run = function (contentType, handler, request, reply
   function done (error, body) {
     // We cannot use resource.bind() because it is broken in node v12 and v14
     resource.runInAsyncScope(() => {
+      process.nextTick(emitDestroy, resource)
```

Comment:
Should we call it in the 404 case too?

Reply:
what's the 404 case?

Reply:
Here: https://github.com/fastify/fastify/blob/0e54206d2cb3c5397a4e714ec7b6ac183fbd63e3/lib/contentTypeParser.js#L163

when the request leads to 404, the `done` callback is not called and the `resource` does not emit the destroy event

Reply:
Done, good catch